lib/matplotlib/legend.py

```diff
@@ -409,42 +409,27 @@ def _set_loc(self, loc):
         # _legend_box will draw itself at the location of the return
         # value of the find_offset.
         self._loc_real = loc
-        if loc == 0:
-            _findoffset = self._findoffset_best
-        else:
-            _findoffset = self._findoffset_loc
-
-#       def findoffset(width, height, xdescent, ydescent):
-#           return _findoffset(width, height, xdescent, ydescent, renderer)
-
-        self._legend_box.set_offset(_findoffset)
-
-        self._loc_real = loc
         self.stale = True
 
     def _get_loc(self):
         return self._loc_real
 
     _loc = property(_get_loc, _set_loc)
 
-    def _findoffset_best(self, width, height, xdescent, ydescent, renderer):
```

Comment:
The idea was to fix the line a bit below (`if iterable(self._loc) and len(self._loc) == 2`) but as you can see the new logic is just to see whether `_loc` is a 0-9 location and otherwise unpack it, so it seemed reasonable to squash the `if ... elif ... else` into a single function rather than keeping them artificially separated.